### ISSUE TITLE
Error: afterSave for _User has already been registered

### ISSUE THREAD

**Issue by droidster** [FirebaseExtended/parse-migration-tools#5]
I'm getting this error when I try to deploy the cloud code: 

```
Error: afterSave for _User has already been registered
    at Object.Parse.Cloud.afterSave (<anonymous>:440:5)
    at migrator.js:101:25
    at Function.T.each.T.forEach (underscore.js:17:779)
    at Migrator.exportTriggers (migrator.js:95:5)
    at main.js:140:10
```

The 140th line being:
` migrator.exportTriggers();`

I have copied migrator.js, firebase-rest.js and consts.js and added this code to my main.js:

```
var migrator = require('cloud/migrator.js')(Parse);
var ref = require('cloud/firebase-rest.js')({
  url: '<url>',
  key: '<i_dont_know_where_to_find_this_key>',
  parse: Parse
});

 migrator.migrateObject(Parse.User, function(object) {
  // Testing
   return ref.child("/users/").child(object.id).put(object);
 });

migrator.exportTriggers();
```

What can be done here? Is this an issue or I have not followed the steps properly?

This project is great but the readme can be improved. As a former parse user, and I must admit I am a bit rusty, I'm finding it really hard to follow as the steps aren't mentioned clearly.

**Comment by droidster** [comment]
Ok, as the error message said afterSave for _User was already registered. My bad! I temporarily removed the afterSave on _User. Where should the afterSave code be written in case of handing migration as well?

**Comment by inlined** [comment]
Migrator has its own beforeSave and afterSave functions that can be used. The library needs do some weird tricks so I had to separate a "logical" before/afterSave from a real one. 

With respect to keys, they've been deprecated in favor of Google's stronger crypto. The param is needed for admin-level access. If you need that, email me at my GitHub handle @ google.com and I'll get you sorted out. Also, check out the latest Firebase CLI release--it now supports importing user auth data!

**Comment by droidster** [comment]
Thanks a lot, that solved it! :) I will email you about bypassing the rules.json. I checked the CLI, it's really helpful, gives us and option to bulk import the data but migration using cloud code seems like the best option for us at will run for 2 more months and keep firebase data updated.
